### PR TITLE
Added a possibility to create `ServiceApiBuilder` with blockchain

### PR DESCRIPTION
**Pull request by aleksuss**
The improvement allows to update Exonum version > 0.8 in EJB.

### CODE REVIEW

File: exonum/src/api/mod.rs [hunk 1]
```diff
@@ -218,14 +220,25 @@ impl ServiceApiScope {
 /// ```
 #[derive(Debug, Clone, Default)]
 pub struct ServiceApiBuilder {
+    blockchain: Option<Blockchain>,
```
Comment (vitvakatu) [inline exonum/src/api/mod.rs]: Any specific reason for `Option` here? Why not storing just `Blockchain`? (with `ServiceApiBuilder::new(b: Blockchain)`)

Comment (aleksuss) [reply exonum/src/api/mod.rs]: @vitvakatu Yes. The reason is to leave a method `new` as is.

File: exonum/src/api/mod.rs [hunk 2]
```diff
@@ -237,6 +250,22 @@ impl ServiceApiBuilder {
...
+    /// Returns an optional reference to the ApiSender.
+    pub fn api_sender(&self) -> Option<&ApiSender> {
+        self.blockchain().map(|blockchain| &blockchain.api_sender)
+    }
+
+    /// Returns an optional value to the PublicKey.
+    pub fn public_key(&self) -> Option<PublicKey> {
+        self.blockchain()
+            .map(|blockchain| blockchain.service_keypair.0)
+    }
```
Comment (ilammy) [inline exonum/src/api/mod.rs]: I'd rephrase the documentation comments to explain when `Some` values are expected from these getters. For example, "Returns service public key of the blockchain if blockchain is set" or something like that. It's obvious from return types that the getters return optional values. Developers are more interested when they can do something with these values.

File: exonum/src/api/mod.rs (hunk 1 above)
Comment (vitvakatu) [reply exonum/src/api/mod.rs]: Ah okay.
@DarkEld3r I think the we might need patch release from 0.9 with this change. It's not a requirement, but it would be much simpler and cleaner solution for EJB team. Is it possible? Otherwise we need to either use git dependency, or wait till 0.10.

File: exonum/src/api/mod.rs (hunk 2 above)
Comment (vldm) [reply exonum/src/api/mod.rs]: @ilammy as for me state: "if blockchain is set" isn't more clear.

Probably we should add info, that this `Option` is safe to unwrap inside `wire` method?

Comment (aleksuss) [reply exonum/src/api/mod.rs]: @vldm I suppose it can be safe if `builder.blockchain().is_some()`

File: exonum/src/api/mod.rs (hunk 1 above)
Comment (DarkEld3r) [reply exonum/src/api/mod.rs]: @vitvakatu  It is definitely possible. :+1:
